src/api/c/convolve.cpp

```diff
@@ -228,6 +229,51 @@ af_err af_convolve2(af_array *out, const af_array signal, const af_array filter,
     } CATCHALL;
 }
 
+template<typename T, typename accT>
+inline static af_array convolve2Strided(const af_array &s, const af_array &f, const dim4 stride, const dim4 padding, const dim4 dilation)
+{
+    return getHandle(convolve2<T, accT>(getArray<T>(s), castArray<accT>(f), stride, padding, dilation));
+}
+
+af_err af_convolve2_strided(af_array *out, const af_array signal, const af_array filter,
+                            const unsigned stride_dims,   const dim_t *strides,
+                            const unsigned padding_dims,  const dim_t *paddings,
+                            const unsigned dilation_dims, const dim_t *dilations)
+{
+    try {
+        const ArrayInfo& sInfo = getInfo(signal);
+        const ArrayInfo& fInfo = getInfo(filter);
+
+        af::dim4 sDims  = sInfo.dims();
+        af::dim4 fDims  = fInfo.dims();
+
+        const af_dtype signalType  = sInfo.getType();
+
+        dim4 stride(stride_dims, strides);
+        dim4 padding(padding_dims, paddings);
+        dim4 dilation(dilation_dims, dilations);
```

Comment:
I am not sure if any values are allowed, already asked this in the comment I made earlier in the header file.

If all values for `strides`, `paddings` and `dilations` are not allowed, we should check for valid values.